### DISCUSSION TITLE
Scheduler used in the original Paper

### DISCUSSION THREAD

**Discussion by rezafuru** [xiaosu-zhu/McQuic#7] [Q&A]
Hi,

Many thanks for open sourcing the code of your excellent paper. Not only did I find your method very creative and amazing, but the effort in making the code this accessible is admirable. 

I'm currently trying to replicate some results, and I'm having some troubles understanding your training configuration. Specifically, regarding the scheduler:

```
...initial learning rate is set at 2 × 10−3 and annealed to 2 × 10−6 at end with cosine learning rate scheduler for 1,000 epochs
```

The way I understand the instructions, I'd initialize a  `torch.optim.lr_scheduler.CosineAnnealingLR` with `T_max=1000, eta_min=2e-6, last_epoch=1000`.

However, this seems to contradict your example configuration [here](https://github.com/xiaosu-zhu/McQuic/blob/main/configs/example.yaml) where you use a custom implementation of `CosineAnnealingWarmRestarts`. This leads me to the following two questions:

1) Did you use  `CosineAnnealingLR` or `CosineAnnealingWarmRestarts` in your original paper to train your mode? If it is the former, could you verify whether the setup above is correct? If its the latter, could you please tell me what the correct setup is?
2) Is there any purpose in a custom implementation of `CosineAnnealingWarmRestarts`? Specifically, I'm wondering whether something is wrong with PyTorch's implementation of `CosineAnnealingWarmRestarts` that made you re-implement the scheduler

If it isn't too much work, is it possible for you to add the configuration file you used in the experiments from the paper?

Best
Alireza
